kyo-core/shared/src/main/scala/kyo/Channel.scala

```diff
@@ -78,6 +80,24 @@ object Channel:
                 }
             }
 
+        /** Takes [[n]] elements from the channel, asynchronously blocking if necessary.
+          *
+          * @return
+          *   The taken element
+          */
+        def takeExactly(n: Int)(using Frame): Seq[A] < (Abort[Closed] & Async) =
+            Loop[Seq[A], Seq[A], Abort[Closed] & Async](Seq.empty): lastChunk =>
+                val nextN = n - lastChunk.size
+                for
+                    drainFiberUnsafe <- IO.Unsafe(self.drainUpToFiber(nextN))
```

Comment:
Would it be possible to reuse the safe version of `drainUpTo` here?

Reply:
For some reason it won't recognize the safe methods when I try to use them here, presumably because it knows that `Channel[A]` is unsafe. Should I rename the `Unsafe` methods so they don't conflict?

Reply:
You can explicitly access the safe version via the companion object: `Channel.drainUpTo(channel)(max)`

Reply:
Ah good to know! Done.